tests/lax_numpy_test.py

```diff
@@ -1384,6 +1384,34 @@ def np_fun(lhs, rhs):
     self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=False, atol=tol,
                           rtol=tol)
 
+  @parameterized.named_parameters(jtu.cases_from_list(  
+    {"testcase_name": "_{}_{}_assume_unique={}_return_indices={}".format(
+    jtu.format_shape_dtype_string(shape1, dtype1),
+    jtu.format_shape_dtype_string(shape2, dtype2),
+    deg,
+    rcond,
+    full,
+    w,
+    cov),
+    "shape1": shape1, "dtype1": dtype1, "shape2": shape2, "dtype2": dtype2, "deg": deg,
+    "rcond": rcond, "full": full, "w":w, "cov":cov}
+    for dtype1 in [s for s in default_dtypes if s != jnp.bfloat16]
+    for dtype2 in [s for s in default_dtypes if s != jnp.bfloat16]
+    for shape1 in all_shapes
+    for shape2 in all_shapes
+    for deg in jnp.array([1, 2, 3, 4, 5])
+    for rcond in [None]
+    for full in [False, True]
+    for w in [None]
+    for cov in [False, True]))
+  def testpolyfit(self, shape1, dtype1, shape2, dtype2, deg, rcond, full, w, cov):
+    rng = jtu.rand_default(self.rng())
+    args_maker = lambda: [rng(shape1, dtype1), rng(shape2, dtype2)]
+    jnp_fun = lambda x, y: jnp.polyfit(x, y, deg=deg, rcond=rcond, full=full, w=w, cov=cov)
+    np_fun = lambda x, y: np.polyfit(x, y, deg=deg, rcond=rcond, full=full, w=w, cov=cov)
+    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=False)
+
```

Comment:
We should add a `self._CompileAndCheck` as well. You'll have to make the other changes suggested above in order for it to pass.